RFC - Security Process Update

j01tz opened the pull request:
Initial WIP for an RFC describing changes to improve Grin's Security Process by adopting a community vulnerability disclosure standard.

[Rendered link to RFC document](https://github.com/j01tz/grin-rfcs/blob/update-security/text/0000-security-process.md)
Tracking issue: https://github.com/mimblewimble/grin-pm/issues/178

lehnberg commented:
Thanks for submitting! I can be shepherd for this one. Does it make sense to then assign it to me?

Also: @j01tz, is this ready for a first review by the community, or are you still working on it?

j01tz commented:
@lehnberg that sounds good, thanks. I'd like to maybe take one more pass today then I'll mark as ready for review.

j01tz commented:
I went back and forth on adding the canaries section. It may be overkill but I thought it was at least worth mentioning for consideration.

quentinlesceller commented on text/0000-security-process.md:
Can this sentence be simplified? Or divided in several sentences?

j01tz replied:
Tried breaking it down a bit. It may warrant some more explanation. Let me know if you think more details should be added after the https://github.com/mimblewimble/grin-rfcs/pull/13/commits/6b0aee02c7d5db79a140380c5011374df842737b update

quentinlesceller replied:
Looking good :+1:.

quentinlesceller commented on text/0000-security-process.md:
What do you mean here?

j01tz replied:
Updated some details for those questions in https://github.com/mimblewimble/grin-rfcs/pull/13/commits/6b0aee02c7d5db79a140380c5011374df842737b. 

> How can we further reduce centralization risk for Grin's security process?

Essentially there are always ways to further reduce centralization risk, it is never a complete process.

>How can we responsibly handle vulnerabilities in academic papers?

A vulnerability could be discovered at specification level in an assumption in a paper we rely on for security etc. That is a bit of a trickier disclosure and not as straightforward as updating code in a repo or adding a banner on a website. 

Thanks very much for reviewing, let me know if you'd like more clarification here.

quentinlesceller replied:
Okay got it thanks for the clarification!

lehnberg commented on text/0000-security-process.md:
In addition, I would consider adding a section in `SECURITY.md` that links to the specific areas of the [standard](https://github.com/RD-Crypto-Spec/Responsible-Disclosure#the-standard) in order to encourage readers to actually review those areas. Specific sections that are particularly important:
* Initial contact
* Giving details
* Setting dates

This might partially replace some of the [Vulnerability handling]( https://github.com/mimblewimble/grin/blob/09cf6de1d143ffbe007478372dc573213e06804d/SECURITY.md#vulnerability-handling), but I think it's important that there's a clear process flow that is easy to review with no conflicting messaging.

j01tz replied:
Great feedback here and overall. I tried to add links to those in the relevant sections. Please let me know if it looks like I missed anything or it doesn't feel intuitive (we want people with vulnerabilities to be able to quickly and clearly see the right thing to do and know how to do it).

j01tz commented on text/0000-security-process.md:
```diff
@@ -0,0 +1,270 @@
+_The standard that this RFC proposes to adopt for Grin is located at https://github.com/RD-Crypto-Spec/Responsible-Disclosure.
+Below we will describe how the standard would be adopted by Grin's Security Process
+in SECURITY.md, based on the example used by Zcash [3]._
+
```
I broke this up a bit. Mentioned the Zcash example more explicitly in the community and reference explanations.

Added a `Changes to SECURITY.md` section and tried refactoring the details to make them more clear with bulleting and links. 

Great feedback, hopefully this was a step in the right direction.

j01tz commented on text/0000-security-process.md:
```diff
@@ -0,0 +1,270 @@
+Details for 'Code Reviews and Audits' can be kept track of on project forums or
+in the 'grin-pm' repo or relevant sub-team repo. The status and resource availability
+for code reviews and audits will change over time and will likely be changing
+too often to be stable enough to be included in a formal security policy.
```
I updated with a statement that the section would be moved to a new page on the wiki but I have not net created and linked a new wiki page.

j01tz commented on text/0000-security-process.md:
```diff
@@ -0,0 +1,270 @@
+
+Likewise 'Chain Splits' is not defined well enough and would probably be better
+tracked and managed outside of the formal security policy, or with a sub-team,
+at least until the language and tooling are better defined and stable.
```
I updated with a statement that the section would be moved to a github issue but I have not yet created and linked the issue. I'll wait to hear from others before I add more details here. 

The tool would be nice to have but perhaps challenging with current resource level. A github issue sounds fine to me for now if it is something that makes sense to keep on the radar.

j01tz commented:
I refactored a lot of the `Reference-level explanation` section to attempt to add more clarity and added a bunch of relevant links to sections in the standard. 

The feedback so far is very much appreciated, please keep it coming!

lehnberg commented:
🎉 Wohooo! This RFC has now been merged! 🤸‍♀️
Tracking issue: https://github.com/mimblewimble/grin-pm/issues/178
